transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -539,16 +539,19 @@ private static void checkMultiplicity(ChannelHandlerContext ctx) {
     }
 
     private void callHandlerAdded(final AbstractChannelHandlerContext ctx) {
-        if (isExecuteLater(ctx)) {
-            ctx.executor().execute(new OneTimeTask() {
-                @Override
-                public void run() {
-                    callHandlerAdded0(ctx);
-                }
-            });
-            return;
+        EventExecutor executor = ctx.executorSafe();
+        if (executor != null) {
```

Comment:
Check the safeExecutor() method ;) 
Basicaly when no special ChannelInvoker/EventExecutorGroup was defined and the Channel was not registered yet